src/shell/validate_utils.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
you can consider remove the header and rename the `validate_utils.cpp`  to `validate_utils.h directly`

Reply:
I don't think it is a good idea. use `.h` and predeclare and avoid files which include `validate_utils.h` to dependency with `command_executor.h`